vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/implementation/VLogReasoner.java

```diff
@@ -808,4 +812,46 @@ ReasonerState getReasonerState() {
 	void setReasonerState(ReasonerState reasonerState) {
 		this.reasonerState = reasonerState;
 	}
+
+	@Override
+	public void writeInferences(OutputStream stream) throws IOException {
+		Set<Predicate> toBeQueriedHeadPredicates = new HashSet<Predicate>();
+		for (Rule rule : this.knowledgeBase.getRules()) {
+			for (Literal literal : rule.getHead()) {
+				toBeQueriedHeadPredicates.add(literal.getPredicate());
+			}
+		}
+		for (DataSourceDeclaration dataSourceDeclaration : this.knowledgeBase.getDataSourceDeclarations()) {
+			toBeQueriedHeadPredicates.add(dataSourceDeclaration.getPredicate());
+		}
+		for (Fact fact : this.knowledgeBase.getFacts()) {
+			toBeQueriedHeadPredicates.add(fact.getPredicate());
+		}
+
+		for (Predicate predicate : toBeQueriedHeadPredicates) {
+			List<Term> tobeGroundedVariables = new ArrayList<Term>();
+			for (int i = 0; i < predicate.getArity(); i++) {
+				tobeGroundedVariables.add(Expressions.makeUniversalVariable("X" + i));
+			}
+			try (final QueryResultIterator answers = this
+					.answerQuery(Expressions.makePositiveLiteral(predicate, tobeGroundedVariables), true)) {
+				answers.forEachRemaining(queryAnswer -> {
+					try {
+						stream.write(Serializer.getFactString(predicate, queryAnswer.getTerms()).getBytes());
+					} catch (IOException e) {
+						throw new RuntimeException(e);

```

Comment:
is this exception turned into an unchecked exception because it is in a forEachRemaning function? Then, it is better to use a non-java8 way of iterating, so that the checked exception can be directly propagated.

Reply:
this comment was not addressed.